service-seed-without-router/all-actors/src/main/java/org/sunbird/BaseActor.java

```diff
@@ -1,35 +1,46 @@
 package org.sunbird;
 
 import akka.actor.UntypedAbstractActor;
-import org.apache.log4j.Logger;
+import akka.event.DiagnosticLoggingAdapter;
+import akka.event.Logging;
 import org.sunbird.message.IResponseMessage;
 import org.sunbird.message.Localizer;
 import org.sunbird.message.ResponseCode;
 import org.sunbird.request.Request;
 
+import java.util.HashMap;
 import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
 
 /**
  * @author Amit Kumar
  */
 public abstract class BaseActor extends UntypedAbstractActor {
 
-    private Logger logger = Logger.getLogger(BaseActor.class);
+    final DiagnosticLoggingAdapter logger = Logging.getLogger(this);
+
     public abstract void onReceive(Request request) throws Throwable;
+
     protected Localizer localizer = Localizer.getInstance();
 
     @Override
     public void onReceive(Object message) throws Throwable {
+        Map<String, Object> mdc;
+        mdc = new HashMap<>();
+        mdc.put("requestId", UUID.randomUUID().toString());
+        logger.setMDC(mdc);
         if (message instanceof Request) {
             Request request = (Request) message;
             String operation = request.getOperation();
-            logger.info("BaseActor:onReceive called for operation:" + operation);
             try {
-                logger.info(String.format("%s:%s:method started at %s",this.getClass().getSimpleName(),operation,System.currentTimeMillis()));
+                logger.info(String.format("%s:%s:method started at %s", this.getClass().getSimpleName(), operation, System.currentTimeMillis()));
```

Comment:
Remove getClass.getSimpleName and timestamp - logger.info will print the class/file. Just retain this message:
- Started : operation
- Ended : operation
- Exception: operation
And all of these above will have a requestId because MDC is set and is getting cleared in finally block.

Reply:
Done

Reply:
Exception line is still missing.